django/db/backends/oracle/operations.py

```diff
@@ -84,6 +84,8 @@ def date_extract_sql(self, lookup_type, field_name):
         if lookup_type == 'week_day':
             # TO_CHAR(field, 'D') returns an integer from 1-7, where 1=Sunday.
             return "TO_CHAR(%s, 'D')" % field_name
+        elif lookup_type == 'week':
+            return "WEEKOF(%s)" % field_name
```

Comment:
https://docs.oracle.com/cd/B28359_01/olap.111/b28126/dml_functions_2136.htm I looked at this. ``TO_CHAR`` supports more datatypes, on a closer read.